site/app/models/gradeable/AutoGradedVersion.php

```diff
@@ -183,6 +185,52 @@ private function loadSubmissionFiles() {
             }
         }
     }
+    public function getDockerErrorFileExist(): bool {
+        $course_path = $this->core->getConfig()->getCoursePath();
+        $gradeable = $this->graded_gradeable->getGradeable();
+        $submitter_id = $this->graded_gradeable->getSubmitter()->getId();
+        $gradeable_id = $gradeable->getId();
+        $gradeable_version = $this->graded_gradeable->getAutoGradedGradeable()->getActiveVersion();
+
+        $filepath = FileUtils::joinPaths(
+            $this->core->getConfig()->getCoursePath(),
+            "results",
+            $gradeable_id,
+            $submitter_id,
+            $gradeable_version,
+            "logs",
+            "docker_error.json"
+        );
+
+        return file_exists($filepath);
+    }
+    /**
+     * @return array<mixed>|null An array with specific keys and values, or null if the file doesn't exist
+     */
+    public function getDockerErrorFileData() {
```

Comment:
This function should also be marked with a native PHP type declaration in addition to the more in-depth PHPDoc comment.

```suggestion
    /**
     * @return array<mixed>|null An array with specific keys and values, or null if the file doesn't exist
     */
    public function getDockerErrorFileData(): array|null {
```